### PR TITLE
Feature/sw5 10/26 add meals

### PR DESCRIPTION
**Pull request by Dra3e**
# Description
This adds the pages to add meals to the systems, edit meals in the system, access the camera and add ingredients to the meal. This also includes the models and services to get the information for those pages.

Fixes #26

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?
The code itself has been run using harcoded data and we have written unit-tests.

**Development Configuration**

* Flutter version: 3.24.3
* Dart version: 3.5.3

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: foodplanner/lib/services/meal_services.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import 'package:foodplanner/models/packed_ingredient.dart';
+import 'package:http/http.dart' as http;
+
+// Fetches a meal by its ID from the server.
+// Takes an HTTP client and the meal ID as parameters.
+/// Returns a Meal object if successful, or throws an exception if not.
+Future<Meal> fetchMeal(http.Client client, int id) async {
+  // Making a GET request to the API to fetch meal details by ID.
+  final response =
+      await client.get(Uri.parse('http://127.0.0.1:80/api/Meals/Get/$id'));
```
Comment (JacobStokholms) [inline foodplanner/lib/services/meal_services.dart]: I må gerne bruge vores $apiUrl, så vi kan streamline URLl i development og senere til stage,  prod.
Skal opdateres i alle fetches.

Example fra fetch_auth.dart:
 final response = await http.post(Uri.parse('$apiUrl/api/Users/Login'),

Comment (Dra3e) [reply foodplanner/lib/services/meal_services.dart]: Jeg har rettet det til at bruge baseUrl fra ApiConfig i stedet, jeg kan godt rette det til præcis det samme som de andre har lavet, men det lader ikke til at ændre funktionaliteten synderligt meget, og føler at denne måde at skrive det op på giver bedre mening.

File: foodplanner/lib/pages/add_meal_page.dart [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+/// This is used to manage the page shifting between "meal_form_page.dart", "add_ingredient_page.dart", and "camera_page.dart".
+class AddMealPage extends StatefulWidget {
+  final Meal meal; // The meal object to be managed within this page.
+
+  const AddMealPage({
+    super.key, // Key for the widget, used for maintaining state.
+    this.meal = const Meal(), // Optional Meal object, defaults to a new Meal instance if none is provided.
+  });
+
+  static const String routeName = '/add_meal_page'; // Route name for navigation.
```
Comment (DollyTheSau) [inline foodplanner/lib/pages/add_meal_page.dart]: Use proper routing, see how its done in routes/index.dart and routes/paths.dart

File: foodplanner/lib/pages/add_ingredient_page.dart [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  const AddIngredientPage({
+    super.key,
+    required this.meal, // Required Meal object to pass.
+    required this.ingredients, // Required list of Ingredient objects to pass.
+    required this.onIngredientsUpdated, // Required callback to handle the ingredient list updating.
+    required this.onCamera, // Required callback to handle camera navigation.
+    required this.client,
+  });
+  
+  static const String routeName = '/add_ingredient_page'; // Route name for navigation.
```
Comment (DollyTheSau) [inline foodplanner/lib/pages/add_ingredient_page.dart]: Same as meal page, route it the proper way through index and paths

**Review by DollyTheSau — CHANGES_REQUESTED** [review]
Follow conventions before pushing

**Comment by lundholm73** [comment]
I skal resolve conflicts før vi kigger på den

File: foodplanner/lib/services/food_image_service.dart [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import 'package:foodplanner/api/openapi/lib/api.dart';
+import 'package:foodplanner/auth/auth_provider.dart';
+import 'package:foodplanner/services/api_config.dart';
+import 'package:http/http.dart' as http;
+import 'package:http/http.dart';
+
+/// This method is used to send the inputted image to the back-end.
```
Comment (lundholm73) [inline foodplanner/lib/services/food_image_service.dart]: Slet hvis det ikke skal brugs

File: foodplanner/pubspec.yaml [hunk 5]
```diff
@@ -7,33 +7,54 @@ environment:
...
 dev_dependencies:
-  flutter_test:
-    sdk: flutter
-  openapi_generator: ^6.0.0
-  build_runner: ^2.4.13
-  mockito: ^5.0.16
-  flutter_lints: ^5.0.0
+    flutter_test:
+        sdk: flutter
+    openapi_generator: ^5.0.2
```
Comment (lundholm73) [inline foodplanner/pubspec.yaml]: hvorfor går vi ned i version?

File: foodplanner/test/page_tests/add_ingredient_page_test.mocks.dart [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+// ignore_for_file: no_leading_underscores_for_library_prefixes
+import 'dart:async' as _i3;
+import 'dart:convert' as _i4;
+import 'dart:typed_data' as _i6;
+
+import 'package:http/http.dart' as _i2;
+import 'package:mockito/mockito.dart' as _i1;
+import 'package:mockito/src/dummies.dart' as _i5;
+
+// ignore_for_file: type=lint
```
Comment (lundholm73) [inline foodplanner/test/page_tests/add_ingredient_page_test.mocks.dart:14]: Væk med det right?

File: foodplanner/lib/config/colors.dart [hunk 7]
```diff
@@ -2,10 +2,11 @@ import 'package:flutter/material.dart';
```
Comment (lundholm73) [inline foodplanner/lib/config/colors.dart]: Hvorfor har i været inde og ændre farverne?

Comment (lundholm73) [reply foodplanner/lib/config/colors.dart]: i må lige kigge i filen var lidt dårlig til og kopiere

File: foodplanner/test/page_tests/add_ingredient_page_test.mocks.dart (hunk 6 above)
Comment (Dra3e) [reply foodplanner/test/page_tests/add_ingredient_page_test.mocks.dart:14]: Det er auto-generated mocks filer, så det giver ikke meget mening at fjerne det, da det vil komme tilbage næste gang vi auto-generate

**Review by JacobStokholms — APPROVED** [review]

**Review by DollyTheSau — APPROVED** [review]

**Pull request merged**